src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java

```diff
@@ -0,0 +1,163 @@
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internetshop.dao.BucketDao;
+import mate.academy.internetshop.exceptions.DataProcessingException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.Bucket;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.util.ConnectionUtil;
+import org.apache.log4j.Logger;
+
+@Dao
+public class BucketDaoJdbcImpl implements BucketDao {
+    private static final Logger LOGGER = Logger.getLogger(BucketDaoJdbcImpl.class);
+
+    @Override
+    public Bucket create(Bucket bucket) {
+        String query = "INSERT INTO buckets(user_id) VALUES (?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement
+                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, bucket.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            resultSet.next();
+            bucket.setId(resultSet.getLong(1));
+            addProductToBucket(bucket);
+            LOGGER.info("The user is added into DB successfully");
+            return bucket;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add the user" + e);
+        }
+    }
+
+    @Override
+    public Optional<Bucket> get(Long id) {
+        String query = "SELECT * FROM buckets WHERE bucket_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Bucket bucket = null;
+            while (resultSet.next()) {
+                bucket = getBucketFromResultSet(resultSet);
+                bucket.setProducts(getAllProductsFromBucket(bucket.getId()));
+            }
+            LOGGER.info("The user is get by ID successfully");
+            return Optional.ofNullable(bucket);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get the bucket by its ID" + e);
+        }
+    }
+
+    @Override
+    public List<Bucket> getAll() {
+        String query = "SELECT * FROM buckets";
+        List<Bucket> buckets = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery(query);
+            while (resultSet.next()) {
+                Bucket bucket = getBucketFromResultSet(resultSet);
+                bucket.setProducts(getAllProductsFromBucket(bucket.getId()));
+                buckets.add(bucket);
+            }
+            LOGGER.info("The information about all users is get successfully");
```

Comment:
copy + paste is evil
```suggestion
            LOGGER.info("The information about all buckets were got successfully");
```

Reply:
Це точно.......